openfisca_uk/calibration/reweighting.py

```diff
@@ -0,0 +1,358 @@
+        weights = sim.calc("household_weight", period=year).values
+        modified_weights = tf.nn.relu(
+            weight_modification[year - START_YEAR] + weights
+        )
+        for variable in variables:
+            if variable in statistic_set.aggregate._children:
+                values = sim.calc(variable, period=year).values
+                entity = sim.simulation.tax_benefit_system.variables[
+                    variable
+                ].entity.key
+                household_totals = sim.map_to(values, entity, "household")
+                household_participants = sim.map_to(
+                    (values > 0) * 1, entity, "household"
+                )
+                # Calculate aggregate error
+                agg = tf.reduce_sum(modified_weights * household_totals)
+                target = getattr(statistic_set.aggregate, variable)
+                l += (
+                    AGGREGATE_ERROR_PENALTY
+                    * target
+                    * ((agg / target) - 1) ** 2
+                )
+            if variable in statistic_set.count._children:
+                # Calculate caseload error
+                count = tf.reduce_sum(
+                    modified_weights * household_participants
+                )
+                target = getattr(statistic_set.count, variable)
+                l += (
+                    PARTICIPATION_ERROR_PENALTY
+                    * target
+                    * ((count / target) - 1) ** 2
+                )
+        for region in regions:
+            in_region = household_region == region
+            people_in_region = household_population * in_region
+            population = tf.reduce_sum(people_in_region * modified_weights)
+            target = getattr(statistic_set.populations, region)
+            l += (
+                POPULATION_ERROR_PENALTY
+                * PARTICIPATION_ERROR_PENALTY
+                * target
+                * ((population / target) - 1) ** 2
+            )
+
+        for country in countries:
+            for variable in statistic_set.aggregate_by_country._children:
+                in_country = household_country == country
+                values = sim.calc(variable, period=year).values
+                entity = sim.simulation.tax_benefit_system.variables[
+                    variable
+                ].entity.key
+                household_totals = sim.map_to(values, entity, "household")
+                aggregate = tf.reduce_sum(
+                    modified_weights * in_country * household_totals
+                )
+                target = getattr(
+                    getattr(statistic_set.aggregate_by_country, variable),
+                    country,
+                )
+                l += (
+                    AGGREGATE_ERROR_PENALTY
+                    * target
+                    * ((aggregate / target) - 1) ** 2
+                )
+        # Income tax payers by tax band
```

Comment:
Yeah actually this section could be its own function to break up the code

Reply:
Yeah plus the others, done